src/wait.rs

```diff
@@ -0,0 +1,70 @@
+/// The module contains a wait primitive
+use crate::Client;
+use std::future::Future;
+use std::time::Duration;
+
+const DEFAULT_POOLING_INTERVAL: Duration = Duration::from_millis(500);
+
+///
+#[derive(Debug)]
+pub struct Wait<'a> {
+    waiter: DefaultWait<&'a mut Client>,
+}
+
+impl<'a> Wait<'a> {
+    ///
+    pub fn new(client: &'a mut Client, timeout: Duration) -> Self {
+        Self {
+            waiter: DefaultWait::new(client, timeout, DEFAULT_POOLING_INTERVAL),
+        }
+    }
+
+    pub(crate) async fn until<F, FF, R>(&mut self, condition: F) -> Result<R, Duration>
```

Comment:
I think the whole `Wait` type only really makes sense if users can define their own waiters, which would mean making this method `pub`, no?

Reply:
It was meant to be a `base` implementation on which could be piled different peaces. But I didn't decide at the time it created for users or for lib itself so it just get a smallest scope I was considering.

It may worth it.